packages/shared/lib/services/connection.service.ts

```diff
@@ -985,55 +985,26 @@ class ConnectionService {
               ) => Promise<Result<boolean, NangoError>>)
             | undefined;
     }): Promise<Result<Connection, NangoError>> {
-        if (connectionId === null) {
-            const error = new NangoError('missing_connection');
-
-            return Err(error);
-        }
-
-        if (providerConfigKey === null) {
-            const error = new NangoError('missing_provider_config');
-
-            return Err(error);
-        }
-
-        const { success, error, response: connection } = await this.getConnection(connectionId, providerConfigKey, environment.id);
-
-        if (!success && error) {
-            return Err(error);
-        }
-
-        if (connection === null || !connection.id) {
-            const error = new NangoError('unknown_connection', { connectionId, providerConfigKey, environmentName: environment.name });
-
-            return Err(error);
-        }
-
-        const config: ProviderConfig | null = await configService.getProviderConfig(connection?.provider_config_key, environment.id);
-
-        if (config === null || !config.id) {
-            const error = new NangoError('unknown_provider_config');
-            return Err(error);
-        }
-
-        const provider = getProvider(config?.provider);
+        const provider = getProvider(integration.provider);
         if (!provider) {
             const error = new NangoError('unknown_provider_config');
             return Err(error);
         }
 
+        const copy = { ...connection };
```

Comment:
why a copy?

Reply:
it's modifying the credentials by reference since we don't fetch anymore inside, it's best practice to copy (even though it's a shallow copy)